File: lab2-functionalprimer/python3/func_as_arg.py
def inpFunc(a, b):
    return [i for i in range(a, b+1)] #Create list of ints from 1 to 5, Haskell equivalent [1..5]

def applicatorFunc(inpFunc, a, b, s):
    if s=='s':
        return sum(inpFunc)
    else:
        return sum(inpFunc)/(b -a +1)

print(applicatorFunc(inpFunc(1, 5), 1, 5, 'A'))

File: lab2-functionalprimer/javascript/func_as_arg.js
//Create list of ints from 1 to 5, Haskell equivalent [1..5]
function arrFunc(a, b){
    let arr = [];
    for (let i = a; i<=b; i++) {
        arr.push(i);
    }
    return arr;    
}

function applicatorFunc(arrFunc, a, b, s){
    if(s=='s'){
        const arr = arrFunc(a, b);        
        let sum = arr.reduce((accumulator, currentValue) => accumulator + currentValue, 0);
        return sum;
    }
    else{        
        const arr = arrFunc(a, b);
        let sum = arr.reduce((accumulator, currentValue) => accumulator + currentValue, 0);
        return sum/(b -a +1);
    }
}

let x = applicatorFunc(arrFunc, 1, 5, 's');
console.log(x);